### PR TITLE
Add Router feature to components

### PR DESCRIPTION
**Pull request by okalimbet**
## Is this a fix or a feature?
Feature

## What is the change?
Refactoring a project with Router. 
MoviePreview was change to a class.
Removed button functionalities to come back to homepage and to a single movie page were replaced with Links.
Correct testing for using Router

## What does it fix?
SRP, readability and better organization of the components
Faster user experience through Router
Adds ability to use navigation and bookmark URL addresses 

## Where should the reviewer start?
App.js line 6
MovieCard.js line 3
MoviePreview.js line 1
Movies.js line 14
index.js line 9
And all related testing for each listed component 

## How should this be tested?
Running npm start would display a fully functional website where User can click on any movie and taken to a movie page. By using a closing button or browser arrows, the User can come back to the previous page
Ensure all testing passes using `npm test`

## Relevant Ticket(s)
#27 #26

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,71 +3,41 @@ import Movies from './Movies/Movies';
...
-  }
-
-  handleMovieClick = (id) => {
-    getSingleMovieData(id)
-    .then(data => this.setState({ 
-      movieObject: data.movie, 
-      clickedMovie: id                         
-    }))
-    .catch(error => this.setState({ error }))
-  }
```
Comment (Shawntru) [inline src/App.js:41]: Nice, I like that this API call was moved to MoviePreview

File: src/App.test.js [hunk 2]
```diff
@@ -63,26 +82,33 @@ describe('App', () => {
...
+    const movieCard = await waitFor(() =>
+      screen.getByTestId('moviecard-element-337401')
+    );
+    fireEvent.click(movieCard);
+
+    await waitFor(() => {
+      expect(screen.getByTestId('movie-preview-page')).toBeInTheDocument();
     });
+  });
+  it('should show a App component for / router', async () => {
```
Comment (okalimbet) [inline src/App.test.js:100]: I learned something new today with these tests for sure! Using Route and Paths inside the MemoryRouter were new to me

File: src/App.test.js [hunk 3]
```diff
@@ -1,82 +1,137 @@
...
-  
   beforeEach(() => {
-    getAllMovieData.mockResolvedValueOnce({ 'movies': [{
-      "id": 694919,
-      "poster_path": "https://image.tmdb.org/t/p/original//6CoRTJTmijhBLJTUNoVSUNxZMEI.jpg",
-      "backdrop_path": "https://image.tmdb.org/t/p/original//pq0JSpwyT2URytdFG0euztQPAyR.jpg",
-      "title": "Money Plane",
-      "average_rating": 6.666666666666667,
-      "release_date": "2020-09-29"
+    getAllMovieData.mockResolvedValueOnce({
```
Comment (Shawntru) [inline src/App.test.js:11]: Good change, having two different movies ensures testing is working with multiple sources/relates better to Movies.js layout

File: src/App.test.js [hunk 4]
```diff
@@ -1,82 +1,137 @@
...
-        "Adventure",
-        "Drama",
-        "Fantasy"
-        ],
-        "budget": 200000000,
-        "revenue": 57000000,
-        "runtime": 115,
-        "tagline": "",
-        "average_rating": 4.909090909090909
-      }
```
Comment (Shawntru) [inline src/App.test.js:70]: Good change moving this to beforeEach

File: src/MovieCard/MovieCard.test.js [hunk 5]
```diff
@@ -2,37 +2,25 @@ import React from 'react';
...
-  it('should fire an closing button event', () => {
-    const mockMovieClickEvent = jest.fn()
-    render(<MovieCard 
-      title={ 'Armagedon' }
-      posterPath={ 'https://image.tmdb.org/t/p/original//6CoRTJTmijhBLJTUNoVSUNxZMEI.jpg' }
-      id={ 694919 }
-      key={ 694919 }
-      movieEventClick={ mockMovieClickEvent }
-     />
-    )
```
Comment (Shawntru) [inline src/MovieCard/MovieCard.test.js:30]: Movie preview exit function on click event no longer called, good to remove its testing

**Review by Shawntru — APPROVED** [review]
Solid changes. Router is setup in a coherent manner, UX is fast and responsive.

**Pull request merged**